core/trino-main/src/main/java/io/trino/operator/aggregation/DistinctAccumulatorFactory.java

```diff
@@ -225,7 +225,7 @@ public void addInput(GroupByIdBlock groupIdsBlock, Page page, Optional<Block> ma
             Block distinctMask = work.getResult();
 
             // 3. feed a Page with a new mask to the underlying aggregation
-            GroupByIdBlock groupIds = new GroupByIdBlock(groupIdsBlock.getGroupCount(), filteredWithGroup.getBlock(0));
+            GroupByIdBlock groupIds = (GroupByIdBlock) filteredWithGroup.getBlock(0);
```

Comment:
What is guarantee that `filteredWithGroup.getBlock(0)` will be `GroupByIdBlock`?

In `OrderingGroupedAccumulator` we don't have that guarantee

Reply:
The `groupIdsBlock` is being prepended to the `page` variable and then this page is being processed by `filter` method which preserves the order of blocks. There is no other option but for the first block to be `GroupByIdBlock`